File: $_algotsu-kaizen/problemo-set/container-with-most-water.c
#define MIN(X, Y) (((X) < (Y)) ? (X) : (Y))
#define MAX(X, Y) (((X) > (Y)) ? (X) : (Y))

int maxArea(int* height, int heightSize) {
    int l = 0x0, r = heightSize - 0x1;
    int area = 0x0;

    do {
        int current = MIN(height[l], height[r]) * (r - l);
        area = MAX(area, current);
        height[l] < height[r] ? ++l : --r;
    } while (l < r);

    return area;
}

File: $_algotsu-kaizen/problemo-set/container-with-most-water.rs
impl Solution {
    pub fn max_area(height: Vec<i32>) -> i32 {
        let mut area: i32 = 0x0;
        let (mut l, mut r): (usize, usize) = (0x0, height.len() - 0x1);

        loop {
            let current = std::cmp::min(height[l], height[r]) * (r - l) as i32;
            area = std::cmp::max(area, current);
            if height[l] < height[r] { l += 0x1 } else { r -= 0x1 }
            if l >= r { break }
        }

        area
    }
}
